app/models/lettings_log.rb

```diff
@@ -79,6 +80,10 @@ def form_name
     FormHandler.instance.form_name_from_start_year(collection_start_year, "lettings")
   end
 
+  def clear_invalid_radio_answers!
+    super if startdate_was && @start_year != FormHandler.instance.collection_start_year(startdate_was)
```

Comment:
Will this clear answers if `startdate_was` is nil? Is it possible that start year is not set before other answers?

Reply:
this is an awkwardness yes. hmmm
part of my thinking was that for things like bulk upload we might rather that the user is "allowed" the invalid answer but gets an invalid answer validation, rather than just voiding it and making them revisit. Maybe that's not the most important thought

Reply:
Actually now I fiddle with it that wasn't my thought!
If I change it to:
`super unless startdate_was && @start_year == FormHandler.instance.collection_start_year(startdate_was)`
then I fail tests:
`BulkUpload::Lettings::Year2023::RowParser validations #field_30 when unpermitted values returns an error`
ie tests that check explicitly that if a value is incorrect they get this validation.
I will check with Phil what his knowledge is from the Bulk Upload side, but yes if we don't check when startdate_was is nil we can definitely find a niche edge case where this bug can be reproduced. My sales log tests on this PR refer to a field whose question is in the setup section

Reply:
Please could you update this and add a test case for it?